library/ssl_tls.c

```diff
@@ -1887,16 +1878,60 @@ static int ssl_compute_master( mbedtls_ssl_handshake_params *handshake,
     if( ret != 0 )
     {
         MBEDTLS_SSL_DEBUG_RET( 1, "prf", ret );
-        return( ret );
     }
+    return( ret );
+}
 
-    if( handshake->premaster == mbedtls_platform_zeroize(
-                    handshake->premaster, sizeof(handshake->premaster) ) )
+/*
+ * Compute master secret if needed
+ *
+ * Parameters:
+ * [in/out] handshake
+ *          [in] resume, premaster, extended_ms, calc_verify, tls_prf
+ *          [out] premaster (cleared)
+ * [out] master
+ * [in] ssl: optionally used for debugging and calc_verify
+ */
+static int ssl_compute_master( mbedtls_ssl_handshake_params *handshake,
+                               unsigned char *master,
```

Comment:
@AndrzejKurek , I thought a lot about this feature. I think there is no chance that some user data will be leaked or session will continue if shared secret was not calculated or calculated wrong. So  we can skip this double master secret calculation, it is not adding security there

Reply:
Removed!